app/controllers/steps_controller.rb

```diff
@@ -97,11 +97,6 @@ def create
           status: :ok
         end
       else
-        # On error, delete the newly added files from S3, as they were
-        # uploaded on client-side (in case of client-side hacking of
-        # asset's signature response)
-        Asset.destroy_all(new_assets)
```

Comment:
This was probably needed when we still had `PAPERCLIP_DIRECT_UPLOAD` option, right?